swf/src/read.rs

```diff
@@ -1576,16 +1576,34 @@ impl<'a> Reader<'a> {
                 FillStyle::Color(color)
             }
 
-            0x10 => FillStyle::LinearGradient(self.read_gradient(shape_version)?),
+            0x10 => {
+                if let Some(gradient) = self.read_gradient(shape_version)? {
+                    FillStyle::LinearGradient(gradient)
+                } else {
+                    FillStyle::Color(Color::BLACK)
+                }
+            }
 
-            0x12 => FillStyle::RadialGradient(self.read_gradient(shape_version)?),
+            0x12 => {
+                if let Some(gradient) = self.read_gradient(shape_version)? {
+                    FillStyle::RadialGradient(gradient)
+                } else {
+                    FillStyle::Color(Color::BLACK)
+                }
+            }
 
-            0x13 => FillStyle::FocalGradient {
-                // SWF19 says focal gradients are only allowed in SWFv8+ and DefineShape4,
```

Comment:
Can we keep this comment?

Reply:
Oops sorry, merged before reading this. Will update this later, sure.